app/javascript/src/components/editor/ExpandableCodeblock.svelte

```diff
@@ -0,0 +1,76 @@
+<script>
+  import { tick } from "svelte"
+  import { reset as microlight } from "../../microlight"
+  import Expand from "../icon/Expand.svelte"
+
+  export let fullContentLines
+  export let snippetLineCount = 3
+  export let snippetHighlightedLineIndex = 0
+
+  let expanded = false
+  let snippetStartLineIdx
+  let snippetLines = []
+  $: {
+    snippetLines = []
+
+    snippetStartLineIdx = snippetHighlightedLineIndex - Math.max(0, Math.floor(snippetLineCount / 2))
+    for (let i = snippetStartLineIdx; i < snippetStartLineIdx + snippetLineCount; i++) {
+      snippetLines.push(fullContentLines[i])
+    }
+  }
+
+  async function syntaxHighlight() {
+    await tick()
+    microlight()
+  }
+
+  $: if (expanded || !expanded) syntaxHighlight()
```

Comment:
Instead of the `if`  you could do
```suggestion
  $: syntaxHighlight(expanded)
```